dask/tests/test_tokenize.py

```diff
@@ -388,13 +386,20 @@ def __init__(self, x):
         def __dask_tokenize__(self):
             return self.x
 
+        def hello(self):
+            return "Hello world"
+
     a, b = Foo(1), Foo(2)
     assert tokenize(a) == tokenize(a)
     assert tokenize(a) != tokenize(b)
 
+    assert tokenize(a.hello) == tokenize(a.hello)
+    assert tokenize(a.hello) != tokenize(b.hello)
```

Comment:
Some updates will be needed here and below downstream of #10808

Reply:
I'm not sure if I agree with this assert. I would agree if the method would have an non-static body, e.g. `return f"Hello world from {self.x}"`

Reply:
What you say makes sense for a `@classmethod` or `@staticmethod`, but definitely not for a regular one. We can't do code inspection at runtime to see if the `self` parameter is actually used or not (but I can bet there's a linter out there exactly for this).
I'll investigate in a follow-up if it's straightforward to deal with `@classmethod` and `@staticmethod`.

Reply:
I would expect the code object to look differently if it is using any variables. Maybe I'm wrong. This is just a high level thought and shouldn't be a blocker.

Reply:
> I would expect the code object to look differently if it is using any variables. 

Sure, nobody stops you from decompiling the code into AST and checking if `self` is ever actually used. I really don't think it's wise to do such a thing outside of a static linter...